packages/accounts/src/msca/account/multiOwnerAccount.ts

```diff
@@ -90,27 +131,44 @@ export async function createMultiOwnerModularAccount({
     ]);
   };
 
-  accountAddress = await getAccountAddress({
+  const _accountAddress = await getAccountAddress({
     client,
-    entryPointAddress: entryPoint.address,
-    accountAddress: accountAddress,
-    getAccountInitCode,
-  });
-
-  const baseAccount = await toSmartContractAccount({
-    transport,
-    chain,
-    entryPoint,
+    entryPoint: _entryPoint,
     accountAddress,
-    source: `MultiOwnerModularAccount`,
     getAccountInitCode,
-    ...standardExecutor,
-    ...multiOwnerMessageSigner(client, accountAddress, () => signer),
   });
 
+  const baseAccount =
+    _entryPoint.version === "0.6.0"
+      ? await toSmartContractAccount({
+          transport,
+          chain,
+          entryPoint: _entryPoint as EntryPointDef<"0.6.0">,
+          accountAddress: _accountAddress,
+          source: `MultiOwnerModularAccount`,
+          getAccountInitCode,
+          ...standardExecutor,
+          ...multiOwnerMessageSigner(client, _accountAddress, () => signer),
+        })
+      : _entryPoint.version === "0.7.0"
+      ? await toSmartContractAccount({
+          transport,
+          chain,
+          entryPoint: _entryPoint as EntryPointDef<"0.7.0">,
+          accountAddress: _accountAddress,
+          source: `MultiOwnerModularAccount`,
+          getAccountInitCode,
+          ...standardExecutor,
+          ...multiOwnerMessageSigner(client, _accountAddress, () => signer),
+        })
```

Comment:
We should look for ways to reduce repetition between the branches. One way to do that might be to factor out all the common props:
```ts
  const commonParams: Omit<
    Parameters<typeof toSmartContractAccount>[0],
    "entryPoint"
  > = {
    transport,
    chain,
    accountAddress: _accountAddress,
    source: `MultiOwnerModularAccount`,
    getAccountInitCode,
    ...standardExecutor,
    ...multiOwnerMessageSigner(client, _accountAddress, () => signer),
  };

  const baseAccount =
    _entryPoint.version === "0.6.0"
      ? await toSmartContractAccount({
          ...commonParams,
          entryPoint: _entryPoint as EntryPointDef<"0.6.0">,
        })
      : _entryPoint.version === "0.7.0"
      ? await toSmartContractAccount({
          ...commonParams,
          entryPoint: _entryPoint as EntryPointDef<"0.6.0">,
        })
      : undefined;
```
(But as Moldy said, this particular account should be 0.6.0 only.)

Reply:
Actually wait a sec, isn't this just the same value in both cases? Can you do
```ts
  const baseAccount = await toSmartContractAccount({
    transport,
    chain,
    entryPoint: _entryPoint as EntryPointDef<"0.6.0"> | EntryPointDef<"0.7.0">,
    accountAddress: _accountAddress,
    source: `MultiOwnerModularAccount`,
    getAccountInitCode,
    ...standardExecutor,
    ...multiOwnerMessageSigner(client, _accountAddress, () => signer),
  });
```

Reply:
modular account doesn't support 0.7 either does it? so we shouldn't be doing this here. 

the default entrypoint version type in https://github.com/alchemyplatform/aa-sdk/pull/541 is the union which means that if the code is the same for both then you wouldn't even need the typecast and you'd be able to leave this file as is without any changes